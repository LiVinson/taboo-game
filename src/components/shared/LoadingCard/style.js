import styled from 'styled-components'

export const StyledLoadingCard = styled.div`
	position: absolute;
	left: 50%;
	top: 50%;
    transform: translate(-50%, -50%);
    z-index: 1000;
`
