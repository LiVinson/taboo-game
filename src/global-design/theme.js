const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
