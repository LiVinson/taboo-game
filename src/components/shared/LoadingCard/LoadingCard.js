import React from 'react'
import PropTypes from 'prop-types'
import { TabooCardTop } from 'components/shared/TabooCard'
import Pending from 'components/shared/Pending'
import { StyledLoadingCard } from './style'

const LoadingCard = ({ message }) => {
	return (
		<StyledLoadingCard>
			<TabooCardTop>
				<Pending message={message} large={true} />
			</TabooCardTop>
		</StyledLoadingCard>
	)
}

LoadingCard.defaultProps = {
	message: 'Loading',
}

LoadingCard.propTypes = {	
	message: PropTypes.string.isRequired,
}

export default LoadingCard
