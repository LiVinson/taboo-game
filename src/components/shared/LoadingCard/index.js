import LoadingCard from "./LoadingCard"

export default LoadingCard