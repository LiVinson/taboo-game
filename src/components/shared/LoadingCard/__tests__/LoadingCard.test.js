import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../../global-design/theme'
import LoadingCard from '../LoadingCard'

describe('LoadingCard', () => {
	const defaultProps = {
		message: 'Fetching deck',
	}

	test('renders correctly', () => {
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<LoadingCard {...defaultProps} />
				</ThemeProvider>
			)
			.toJSON()
		expect(wrapper).toMatchSnapshot()
	})
})
