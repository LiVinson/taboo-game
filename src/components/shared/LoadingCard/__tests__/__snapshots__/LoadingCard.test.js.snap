// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingCard renders correctly 1`] = `
.c1 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c1:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c3 {
  padding: 3.5rem .5rem;
  font-size: 3.5rem;
  color: #333333;
}

.c0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 1000;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <p
        className="c3"
      >
        Fetching deck
      </p>
    </div>
  </div>
</div>
`;
